compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1268,6 +1268,24 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
 
         if return_span != borrow_span {
             err.span_label(borrow_span, note);
+
+            let tcx = self.infcx.tcx;
+
+            let suggestion = match tcx.sess.source_map().span_to_snippet(return_span) {
+                Ok(string) => string.contains(".iter()") && !string.contains(".collect"),
+                Err(_) => false,
+            };
+
+            // If the snippet contains ".iter()", then
+            // make `.collect()` suggestion.
+            if suggestion {
+                err.span_suggestion_short(
+                    borrow_span,
+                    "Maybe use `.collect()` to allocate the iterator",
```

Comment:
I wouldn't say "maybe"; hints are already assumed to not always be accurate.
```suggestion
                    "use `.collect()` to allocate the iterator",
```

Reply:
Thank you, that was changed